### PR TITLE
Convert all references to lowercase

### PR DESCRIPTION
**Pull request by danimtb**
In order to avoid problems with case-sensitive OSs, we can start using only lowercase references in our documentation and examples in order to void them.

Still, we are not enforcing lowercase usage in Conan, but this could be a good de-facto practice in our documentation to start teaching the users,

### CODE REVIEW

File: developing_packages/workspaces.rst [hunk 1]
```diff
@@ -118,11 +118,11 @@ depend on different versions of the same library, as in any other Conan command.
 
     root: chat/0.1@user/testing, say/0.1@user/testing
     # or
-    root: ["HelloA/0.1@lasote/stable", "HelloB/0.1@lasote/stable"]
+    root: ["helloA/0.1@lasote/stable", "helloB/0.1@lasote/stable"]
```
Comment (danimtb) [inline developing_packages/workspaces.rst]: check this

File: developing_packages/workspaces.rst [hunk 2]
```diff
@@ -118,11 +118,11 @@ depend on different versions of the same library, as in any other Conan command.
...
     # or
     root:
-        - HelloA/0.1@lasote/stable
-        - HelloB/0.1@lasote/stable
+        - helloA/0.1@lasote/stable
+        - helloB/0.1@lasote/stable
```
Comment (danimtb) [inline developing_packages/workspaces.rst]: and this

File: howtos/manage_shared_libraries/env_vars.rst [hunk 3]
```diff
@@ -67,42 +67,42 @@ you will need to have the shared library available.
...
     class PackageB(ConanFile):
-        name = "packageB"
+        name = "package_b"
         version = "1.0"
-        requires = "toolA/1.0@myuser/stable"
+        requires = "tool_a/1.0@myuser/stable"
 
         def build(self):
-            exe_name = "toolA.exe" if self.settings.os == "Windows" else "toolA"
+            exe_name = "tool_a.exe" if self.settings.os == "Windows" else "tool_a"
```
Comment (danimtb) [inline howtos/manage_shared_libraries/env_vars.rst]: check executable name

File: developing_packages/workspaces.rst (hunk 1 above)
Comment (memsharded) [reply developing_packages/workspaces.rst]: Yes, I suggest moving everything strictly to lowercase ``helloa`` or ``hello_a``

File: howtos/manage_shared_libraries/env_vars.rst (hunk 3 above)
Comment (danimtb) [reply howtos/manage_shared_libraries/env_vars.rst]: It is ok

File: howtos/manage_shared_libraries/env_vars.rst [hunk 4]
```diff
@@ -175,6 +175,6 @@ In the terminal window:
 
     $ conan install .
     $ source activate_run
-    $ toolA --someparams
+    $ toola --someparams
```
Comment (danimtb) [inline howtos/manage_shared_libraries/env_vars.rst]:
```suggestion
    $ tool_a --someparams
```

File: reference/commands/output/info.rst [hunk 5]
```diff
@@ -34,18 +34,17 @@ each nested one can be built in parallel.
...
-                "LibF/0.1@lasote/stable"
+                "liba/0.1@lasote/stable",
+                "libe/0.1@lasote/stable",
+                "libf/0.1@lasote/stable"
             ],
             [  
-                "LibB/0.1@lasote/stable",
-                "LibC/0.1@lasote/stable"
+                "libb/0.1@lasote/stable",
+                "libc/0.1@lasote/stable"
```
Comment (czoido) [inline reference/commands/output/info.rst]: Shouldn't these references be converted to snake case as well?

File: reference/commands/output/upload.rst [hunk 6]
```diff
@@ -92,7 +92,7 @@ The output JSON contains a two first level keys:
             },
             {
                 "recipe":{
-                    "id":"HelloApp/0.1@conan/testing",
+                    "id":"hello_app/0.1@conan/testing",
```
Comment (czoido) [inline reference/commands/output/upload.rst]: could be this?
```suggestion
                    "id":"helloapp/0.1@conan/testing",
```

**Comment by danimtb** [comment]
I don't care that much about the style. For me, the snake-case is more readable and the lower-case is more *writeable*...

File: creating_packages/define_abi_compatibility.rst [hunk 7]
```diff
@@ -312,36 +310,33 @@ In the above example, we will transform the package ID of the ``Visual Studio``
...
 Dependency Issues
 -----------------
 
-Let's define a simple scenario whereby there are two packages: ``MyOtherLib/2.0`` and ``MyLib/1.0`` which depends on
-``MyOtherLib/2.0``. Let's assume that their recipes and binaries have already been created and uploaded to a Conan remote.
+Let's define a simple scenario whereby there are two packages: ``my_other_lib/2.0`` and ``my_lib/1.0`` which depends on
+``my_other_lib/2.0``. Let's assume that their recipes and binaries have already been created and uploaded to a Conan remote.
 
-Now, a new release for ``MyOtherLib/2.1`` is released with an improved recipe and new binaries. The ``MyLib/1.0`` is modified and is required to be upgraded to ``MyOtherLib/2.1``.
+Now, a new release for ``my_other_lib/2.1`` is released with an improved recipe and new binaries. The ``my_lib/1.0`` is modified and is required to be upgraded to ``MyOtherLib/2.1``.
```
Comment (jgsogo) [inline creating_packages/define_abi_compatibility.rst]:
```suggestion
Now, a new release for ``my_other_lib/2.1`` is released with an improved recipe and new binaries. The ``my_lib/1.0`` is modified and is required to be upgraded to ``my_other_lib/2.1``.
```

File: creating_packages/define_abi_compatibility.rst [hunk 8]
```diff
@@ -406,10 +401,10 @@ If it is necessary to change the default behavior, the applied versioning schema
...
     class PkgConan(ConanFile):
-        name = "Mylib"
+        name = "my_lib"
         version = "1.0"
         settings = "os", "compiler", "build_type", "arch"
-        requires = "MyOtherLib/2.0@lasote/stable"
+        requires = "my_other_lib/2.0@lasote/stable"
 
         def package_id(self):
             myotherlib = self.info.requires["MyOtherLib"]
```
Comment (jgsogo) [inline creating_packages/define_abi_compatibility.rst]:
```suggestion
            myotherlib = self.info.requires["my_other_lib"]
```

And below this line, there are other `MyOtherLib` that we should change

File: creating_packages/define_abi_compatibility.rst [hunk 9]
```diff
@@ -476,7 +471,7 @@ All the modes can be applied to all dependencies, or to individual ones:
   .. code-block:: text
 
     MyLib/1.2.3@user/testing       => MyLib/1.Y.Z
```
Comment (jgsogo) [inline creating_packages/define_abi_compatibility.rst]:
```suggestion
    my_lib/1.2.3@user/testing       => my_lib/1.Y.Z
```

And above this, there is another code-block that requires a change

File: creating_packages/define_abi_compatibility.rst [hunk 10]
```diff
@@ -476,7 +471,7 @@ All the modes can be applied to all dependencies, or to individual ones:
...
-    MyOtherLib/2.3.4@user/testing  =>
+    my_other_lib/2.3.4@user/testing  =>
 
   So the direct dependencies are mapped to the major version only. Changing its channel, or using version
   ``MyLib/1.4.5`` will still produce ``MyLib/1.Y.Z`` and thus the same package-id.
```
Comment (jgsogo) [inline creating_packages/define_abi_compatibility.rst]:
```suggestion
  ``my_lib/1.4.5`` will still produce ``my_lib/1.Y.Z`` and thus the same package-id.
```

File: creating_packages/define_abi_compatibility.rst [hunk 11]
```diff
@@ -488,21 +483,21 @@ All the modes can be applied to all dependencies, or to individual ones:
...
-      self.info.requires["MyOtherLib"].semver_mode()
+      self.info.requires["my_other_lib"].semver_mode()
 
   This results in:
 
   .. code-block:: text
 
     MyLib/1.2.3@user/testing       => MyLib/1.Y.Z
-    MyOtherLib/2.3.4@user/testing  => MyOtherLib/2.Y.Z
+    MyOtherLib/2.3.4@user/testing  => my_other_lib/2.Y.Z
```
Comment (jgsogo) [inline creating_packages/define_abi_compatibility.rst]:
```suggestion
    my_other_lib/2.3.4@user/testing  => my_other_lib/2.Y.Z
```

File: creating_packages/define_abi_compatibility.rst [hunk 12]
```diff
@@ -488,21 +483,21 @@ All the modes can be applied to all dependencies, or to individual ones:
...
 
     def package_id(self):
-      self.info.requires["MyOtherLib"].semver_mode()
+      self.info.requires["my_other_lib"].semver_mode()
 
   This results in:
 
   .. code-block:: text
 
     MyLib/1.2.3@user/testing       => MyLib/1.Y.Z
```
Comment (jgsogo) [inline creating_packages/define_abi_compatibility.rst]:
```suggestion
    my_lib/1.2.3@user/testing       => my_lib/1.Y.Z
```

File: creating_packages/define_abi_compatibility.rst [hunk 13]
```diff
@@ -535,38 +530,38 @@ All the modes can be applied to all dependencies, or to individual ones:
...
-    MyOtherLib/1.3.4-a4+b3@user/testing  => MyOtherLib/1.3.4-a4+b3@user/testing   
+    my_other_lib/1.3.4-a4+b3@user/testing  => my_other_lib/1.3.4-a4+b3@user/testing   
 
 - ``full_package_mode()``: Any change in the required version, user, channel or package ID changes the package ID.
 
   .. code-block:: python
 
     def package_id(self):
-      self.info.requires["MyOtherLib"].full_package_mode()
+        self.info.requires["MyOtherLib"].full_package_mode()
```
Comment (jgsogo) [inline creating_packages/define_abi_compatibility.rst]:
```suggestion
        self.info.requires["my_other_lib"].full_package_mode()
```

File: creating_packages/define_abi_compatibility.rst [hunk 14]
```diff
@@ -721,74 +716,74 @@ a single dependency, which is a static library. Then, downstream
...
 
 
 Library Types: Shared, Static, Header-only
 ++++++++++++++++++++++++++++++++++++++++++
 
 Let's see some examples, corresponding to common scenarios:
 
-- ``MyLib/1.0`` is a shared library that links with a static library ``MyOtherLib/2.0`` package.
+- ``my_lib/1.0`` is a shared library that links with a static library ``my_other_lib/2.0`` package.
   When a new ``MyOtherLib/2.1`` version is released: Do I need to create a new binary for
```
Comment (jgsogo) [inline creating_packages/define_abi_compatibility.rst]:
```suggestion
  When a new ``my_other_lib/2.1`` version is released: Do I need to create a new binary for
```

File: creating_packages/define_abi_compatibility.rst [hunk 15]
```diff
@@ -721,74 +716,74 @@ a single dependency, which is a static library. Then, downstream
...
-  they used in ``MyLib/1.0``. Adding new methods to the public headers will have no impact, but
+  they used in ``my_lib/1.0``. Adding new methods to the public headers will have no impact, but
   changing the implementation of some functions that will be inlined when compiled from
-  ``MyLib/1.0`` will definitely require re-building. For this case, it could make sense to have this configuration:
+  ``my_lib/1.0`` will definitely require re-building. For this case, it could make sense to have this configuration:
 
   .. code-block:: python
 
       def package_id(self):
           # Any change in the MyOtherLib version, user or channel
```
Comment (jgsogo) [inline creating_packages/define_abi_compatibility.rst]:
```suggestion
          # Any change in the my_other_lib version, user or channel
```

File: creating_packages/define_abi_compatibility.rst [hunk 16]
```diff
@@ -721,74 +716,74 @@ a single dependency, which is a static library. Then, downstream
...
           # or Package ID will affect our package ID
-          self.info.requires["MyOtherLib"].full_package_mode()
+          self.info.requires["my_other_lib"].full_package_mode()
 
           # Or any change in the MyOtherLib version, user or
```
Comment (jgsogo) [inline creating_packages/define_abi_compatibility.rst]:
```suggestion
          # Or any change in the my_other_lib version, user or
```

File: devtools/build_requires.rst [hunk 17]
```diff
@@ -57,13 +57,13 @@ Build requirements can also be specified in a package recipe, with the ``build_r
...
+        build_requires = "tool_a/0.2@user/testing", "tool_b/0.2@user/testing"
 
         def build_requirements(self):
             # useful for example for conditional build_requires
             # This means, if we are running on a Windows machine, require ToolWin
             if platform.system() == "Windows":
-                self.build_requires("ToolWin/0.1@user/stable")
+                self.build_requires("tool_win/0.1@user/stable")
 
 The above ``ToolA`` and ``ToolB`` will always be retrieved and used for building this recipe, while the ``ToolWin`` one will only be used
```
Comment (jgsogo) [inline devtools/build_requires.rst]:
```suggestion
The above ``tool_a`` and ``tool_b`` will always be retrieved and used for building this recipe, while the ``tool_win`` one will only be used
```

File: howtos/cmake_launch.rst [hunk 18]
```diff
@@ -42,4 +42,4 @@ If you want to use targets, you could do:
                     BASIC_SETUP CMAKE_TARGETS)
```
Comment (jgsogo) [inline howtos/cmake_launch.rst]: Use `catch2` from Conan Center? (for a different issue/PR)

Comment (danimtb) [reply howtos/cmake_launch.rst]: we have to review all the examples from the documentation so...

**Comment by danimtb** [comment]
Thanks @jgsogo for all the suggestions. As you said, it is very difficult to catch them all at once, so let's push this forward and we can discuss later the casing or make another review to the whole docs

**Pull request merged**